src/Interpreters/InterpreterDeleteQuery.cpp

```diff
@@ -75,39 +75,45 @@ BlockIO InterpreterDeleteQuery::execute()
+        command->predicate = delete_query.predicate;
+        command->update_assignments = std::make_shared<ASTExpressionList>();
+        auto set_row_does_not_exist = std::make_shared<ASTAssignment>();
+        set_row_does_not_exist->column_name = LightweightDeleteDescription::FILTER_COLUMN.name;
+        auto zero_value = std::make_shared<ASTLiteral>(DB::Field(UInt8(0)));
+        set_row_does_not_exist->children.push_back(zero_value);
+        command->update_assignments->children.push_back(set_row_does_not_exist);
+        command->children.push_back(command->predicate);
+        command->children.push_back(command->update_assignments);
+        mut_command.column_to_update_expression[set_row_does_not_exist->column_name] = zero_value;
+        mut_command.ast = command->ptr();
+
+        mutation_commands.emplace_back(mut_command);
+
+        table->checkMutationIsPossible(mutation_commands, getContext()->getSettingsRef());
+        MutationsInterpreter(table, metadata_snapshot, mutation_commands, getContext(), false).validate();
+        table->mutate(mutation_commands, getContext());
 
-    if (!getContext()->getSettingsRef().allow_experimental_lightweight_delete)
-        throw Exception(ErrorCodes::SUPPORT_IS_DISABLED, "Lightweight delete mutate is experimental. Set `allow_experimental_lightweight_delete` setting to enable it");
-
-    /// Convert to MutationCommand
-    MutationCommands mutation_commands;
-    MutationCommand mut_command;
-
-    /// Build "UPDATE _row_exists = 0 WHERE predicate" query
-    mut_command.type = MutationCommand::Type::UPDATE;
-    mut_command.predicate = delete_query.predicate;
-
-    auto command = std::make_shared<ASTAlterCommand>();
-    command->type = ASTAlterCommand::UPDATE;
-    command->predicate = delete_query.predicate;
-    command->update_assignments = std::make_shared<ASTExpressionList>();
-    auto set_row_does_not_exist = std::make_shared<ASTAssignment>();
-    set_row_does_not_exist->column_name = LightweightDeleteDescription::FILTER_COLUMN.name;
-    auto zero_value = std::make_shared<ASTLiteral>(DB::Field(UInt8(0)));
-    set_row_does_not_exist->children.push_back(zero_value);
-    command->update_assignments->children.push_back(set_row_does_not_exist);
-    command->children.push_back(command->predicate);
-    command->children.push_back(command->update_assignments);
-    mut_command.column_to_update_expression[set_row_does_not_exist->column_name] = zero_value;
-    mut_command.ast = command->ptr();
-
-    mutation_commands.emplace_back(mut_command);
-
-    table->checkMutationIsPossible(mutation_commands, getContext()->getSettingsRef());
-    MutationsInterpreter(table, metadata_snapshot, mutation_commands, getContext(), false).validate();
-    table->mutate(mutation_commands, getContext());
-
-    return {};
+        return {};
+    }
+    else
+    {
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "DELETE query is not supported for table");
```

Comment:
```suggestion
        throw Exception(ErrorCodes::BAD_ARGUMENTS, "DELETE query is not supported for tables with {} engine", table->getName());
```

Reply:
I thought about it, but the thing is that  table->getName() returns storage type (for example, "StorageMergeTree") and not the table name. Then the error message might confuse the user because supportDeletes() might return false or true not only based on storage type but also depending on the schema of the actual table or depending on whether it has some indexes or projections.

Reply:
In this case, it seems like `table->getStorageID()` can be used

Reply:
Added table name, as you suggested: https://github.com/ClickHouse/ClickHouse/pull/44806